src/main/java/Dackel.java

```diff
@@ -0,0 +1,382 @@
+     */
+    private static void addDeadline(String taskName, String dueTime) {
+        Deadline newTask = new Deadline(taskName, dueTime);
+        storedTasks.add(newTask);
+        numberOfTasks++;
+        speak("added the following task to your list!\n " + newTask.toString());
+        speak("your list now has " + String.valueOf(numberOfTasks) + " tasks.");
+    }
+
+    /**
+     * Adds an Event to storedTasks
+     * 
+     * @param taskName name of task to be added
+     * @param startTime starting date/time of task as a String
+     * @param endTime ending date/time of task as a String
+     */
+    private static void addEvent(String taskName, String startTime, String endTime) {
+        Event newTask = new Event(taskName, startTime, endTime);
+        storedTasks.add(newTask);
+        numberOfTasks++;
+        speak("added the following task to your list!\n " + newTask.toString());
+        speak("your list now has " + String.valueOf(numberOfTasks) + " tasks.");
+    }
+
+    /**
+     * Removes task with specified index from storedTasks
+     * 
+     * @param index index of the element to be removed
+     */
+    private static void deleteTask(int index) {
+        Task taskToBeRemoved = storedTasks.get(index);
+        String s = "the following task will be deleted from your list:\n ";
+        s += taskToBeRemoved.toString();
+        speak(s);
+        speak("are you sure you want to do this? [Y/n]: ");
+        String confirmation = receiveInput();
+        if (confirmation.equals("Y")) {
+            storedTasks.remove(index);
+            numberOfTasks--;
+            speak("task " + taskToBeRemoved.toString() + " successfully deleted.");
+            speak("you now have " + String.valueOf(numberOfTasks) + " tasks on your list.");
+        }
+        else {
+            speak("task was not deleted.");
+        }
+    }
+
+    /**
+     * Lists all tasks in storedTasks. Displays a message if storedTasks is empty
+     */
+    private static void listTasks() {
+        if (numberOfTasks == 0) {
+            speak("you have no tasks on your list!");
+            return;
+        }
+        String s = "";
+        for (int i = 0; i < numberOfTasks; i++) {
+            s += "\n" + String.format(" %2d " + storedTasks.get(i).toString(), i + 1);
+        }
+        speak(s);
+    }
+
+    /**
+     * Marks the specified task as done
+     * 
+     * @param index index of task in storedTasks
+     */
+    private static void markTask(int index) {
+        storedTasks.get(index).mark();
+        speak("i've marked the following task as done!\n " + storedTasks.get(index).toString());
+    }
+
+    /**
+     * Marks the specified task as not done
+     * 
+     * @param index index of task in storedTasks
+     */
+    private static void unmarkTask(int index) {
+        storedTasks.get(index).unmark();
+        speak("i've unmarked the following task. do it soon, please!\n " + storedTasks.get(index).toString());
+    }
```

Comment:
I think splitting up the functionality of this class into separate classes would be better? That way, implementing new functionality would be easier and it is also easier to debug if there are errors.